src/gridfs-stream/upload.ts

```diff
@@ -104,9 +125,14 @@ export class GridFSBucketWriteStream extends Writable {
    * @returns {boolean} False if this write required flushing a chunk to MongoDB. True otherwise.
    */
 
-  write(chunk: any, encoding: any, callback?: Callback): boolean {
-    var _this = this;
-    return waitForIndexes(this, () => doWrite(_this, chunk, encoding, callback));
+  write(
```

Comment:
Same as the comment about overloads above. Do we need to add overloads if something always returns the same type? Please clarify what the idea is here.

Reply:
The one above was fine because the last parameter was just an optional. In this case we need overloads because we want to guide the user's completion:
```
write(chunk: Buffer, callback: Callback<void>);
write(chunk: Buffer, encoding: BufferEncoding, callback: Callback<void>);
.. so on ..
```
Overloading isn't just about the return type